### PR TITLE
Add .NET GitHub Action reference

### LINKED ISSUE #25194: Document code-analysis GitHub action
Add an article about official dotnet (first-party) GitHub actions [here](https://docs.microsoft.com/en-us/dotnet/devops/github-actions-overview). Document the following actions:

- [ ] [dotnet/code-analysis](https://github.com/dotnet/code-analysis) - For this one, also add a backlink from https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview.

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #25194.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/devops/dotnet-github-action-reference?branch=pr-en-us-26183).

### CODE REVIEW

**Comment by mavasani** [comment]
LGTM.

@mikadumont may want to review before we publish it.

**Review by IEvangelist — APPROVED** [review]
Approved with one nit

File: docs/devops/dotnet-github-action-reference.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+## .NET version sweeper
+
+[dotnet/versionsweeper](https://github.com/dotnet/versionsweeper)
+
+This action sweeps .NET repos for out-of-support target versions of .NET.
+
+The .NET docs team uses the .NET version sweeper GitHub Action to automate issue creation. The Action runs on a schedule (as a cron job). When it detects that .NET projects target out-of-support versions, it creates issues to report its findings. The output is configurable and helpful for tracking .NET version support concerns. The Action is available on [GitHub Marketplace](https://github.com/marketplace/actions/net-version-sweeper).
+
+## .NET code analysis GitHub Action
```
Comment (adegeo) [inline docs/devops/dotnet-github-action-reference.md:22]: This heading has `GitHub Action` which repeats the H1 and doesn't match the previous H2 heading, I suggest it's removed
```suggestion
## .NET code analysis
```

File: docs/devops/dotnet-github-action-reference.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+> [!NOTE]
+> This article is a work-in-progress, and may not list all the available .NET GitHub Actions.
+
+## .NET version sweeper
+
+[dotnet/versionsweeper](https://github.com/dotnet/versionsweeper)
+
+This action sweeps .NET repos for out-of-support target versions of .NET.
+
+The .NET docs team uses the .NET version sweeper GitHub Action to automate issue creation. The Action runs on a schedule (as a cron job). When it detects that .NET projects target out-of-support versions, it creates issues to report its findings. The output is configurable and helpful for tracking .NET version support concerns. The Action is available on [GitHub Marketplace](https://github.com/marketplace/actions/net-version-sweeper).
```
Comment (adegeo) [inline docs/devops/dotnet-github-action-reference.md:20]: I think putting this on its own line makes how you get it, standout
```suggestion
The .NET docs team uses the .NET version sweeper GitHub Action to automate issue creation. The Action runs on a schedule (as a cron job). When it detects that .NET projects target out-of-support versions, it creates issues to report its findings. The output is configurable and helpful for tracking .NET version support concerns.

The Action is available on [GitHub Marketplace](https://github.com/marketplace/actions/net-version-sweeper).
```

File: docs/devops/dotnet-github-action-reference.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+This action runs the code analysis rules that are included in the .NET SDK as part of continuous integration (CI). The action runs both [code-quality (CAXXXX) rules](../fundamentals/code-analysis/quality-rules/index.md) and [code-style (IDEXXXX) rules](../fundamentals/code-analysis/style-rules/index.md). Consider using this GitHub Action in the following scenarios:
+
+- You only want to see compiler diagnostics when you compile locally, but you still want to catch code analysis issues in a separate phase.
+- You want to improve compile-time performance by offloading expensive analyzers, such as the dataflow analysis-based security analyzers, to the CI phase.
+- You want to run the default .NET SDK code analyzers when you compile locally, but you want to run an extended set of code analyzers in the CI phase.
+
+You can configure the action in various ways, including whether you want violations to break the CI build. For more information, see the [README file](https://github.com/dotnet/code-analysis/blob/main/README.md). For more information about .NET code analysis, see [Overview of .NET code analysis](../fundamentals/code-analysis/overview.md).
```
Comment (adegeo) [inline docs/devops/dotnet-github-action-reference.md:32]: Is there basic information about how someone gets this action? The previous action says its available on the marketplace.

**Review by adegeo — APPROVED** [review]
LGTM I left some comments about some minor things to consider 😄

File: docs/devops/dotnet-github-action-reference.md (hunk 3 above)
Comment (gewarren) [reply docs/devops/dotnet-github-action-reference.md:34]: The first line is a link to the action on github. I don't think this one's on the Marketplace?

File: docs/fundamentals/code-analysis/overview.md [hunk 4]
```diff
@@ -146,6 +146,10 @@ dotnet_diagnostic.CA1822.severity = none
 
 For more information and other ways to suppress warnings, see [How to suppress code analysis warnings](suppress-warnings.md).
 
+## Run code analysis as a GitHub Action
+
+The [dotnet/code-analysis](https://github.com/dotnet/code-analysis) GitHub Action lets you run .NET code analyzers as part of continuous integration (CI) in an offline mode. For more information, see [.NET code analysis GitHub Action](../../devops/dotnet-github-action-reference.md#net-code-analysis-github-action).
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/overview.md:151]:
```suggestion
The [dotnet/code-analysis](https://github.com/dotnet/code-analysis) GitHub Action lets you run .NET code analyzers as part of continuous integration (CI) in an offline mode. For more information, see [.NET code analysis GitHub Action](../../devops/dotnet-github-action-reference.md#net-code-analysis).
```

**Comment by gewarren** [comment]
Thanks for the reviews!

**Pull request merged**